web/src/user_profile.ts

```diff
@@ -366,11 +435,17 @@ export function get_custom_profile_field_data(user, field, field_types) {
     return profile_field;
 }
 
-export function update_user_custom_profile_fields(user) {
+export function update_user_custom_profile_fields(user: User): void {
     if (!(modals.any_active() && modals.active_modal() === "#user-profile-modal")) {
         return;
     }
-    const current_user_id = Number.parseInt(original_values.user_id, 10);
+    if (original_values?.user_id === undefined) {
+        return;
+    }
```

Comment:
Do we expect this case to happen? Wondering if it should be an assert.

Reply:
I thought of using `assert` initially before I came across this commit 5d88ff6676c222220599252b5d47d594e27e4b47
I tried reproducing this and it gives `original_values.user_id` as undefined.

Reply:
Oh, no. I believe since there is no check, this is happening in production too.
If we add this conditional here, the function becomes useless, because probably `original_values.user_id` is always `undefined`

Reply:
I think if you read the code about what `original_values` is for, it's something to do with comparing what has changed to managed the "save changes" button or live update or something.

Reply:
Yes, but I meant that there is a flaw in the logic. There can be few occasions when the `original_values` are undefined.
I've filed the issue in the [CZO](https://chat.zulip.org/#narrow/stream/9-issues/topic/custom_profile_fields.20not.20getting.20updated.20by.20server_events) also, showing a live example of the case when the original_values was undefined.

The reason for that is when we open the profile_modal (with any of the tabs except `manage_user`) of the user the original value is not being set initially, resulting in `original_values` being `undefined`.